File: backend/app/schemas/scholarship.py
from pydantic import BaseModel
from typing import List, Optional


class ScholarshipOut(BaseModel):
    id: str
    name: str
    provider: str
    amount: float
    eligibilityDescription: str
    academicRequirements: str
    incomeRequirements: str
    deadline: str
    requiredDocuments: List[str]
    applicationStatus: Optional[str] = None
    officialUrl: str
    category: Optional[str] = None
    academicLevel: Optional[str] = None

    model_config = {"from_attributes": True}


class EligibilityCheckRequest(BaseModel):
    course: str
    year: int
    cgpa: float
    familyIncome: float
    location: str
    category: Optional[str] = None


class EligibilityResultOut(BaseModel):
    scholarship: dict
    eligibility: dict
    source: dict
    relevance: Optional[float] = None

File: src/types/scholarship.ts
// ── Scholarship types ────────────────────────────────────────────────────────
export type EligibilityStatus =
  | 'likely_eligible'
  | 'may_not_be_eligible'
  | 'more_information_required';

export interface Scholarship {
  id: string;
  name: string;
  provider: string;
  amount: number;
  eligibilityDescription: string;
  academicRequirements: string;
  incomeRequirements: string;
  deadline: string;
  requiredDocuments: string[];
  applicationStatus?: 'not_applied' | 'applied' | 'approved' | 'rejected';
  officialUrl: string;
  category?: string;
  academicLevel?: string;
}

export interface EligibilityCheckRequest {
  course: string;
  year: number;
  cgpa: number;
  familyIncome: number;
  location: string;
  category?: string;
}

export interface EligibilityResult {
  scholarship: {
    name: string;
    provider: string;
    amount: number;
    deadline: string;
  };
  eligibility: {
    status: EligibilityStatus;
    explanation: string;
    matchingCriteria: string[];
  };
  source: {
    title: string;
    url: string;
    publisher: string;
  };
  relevance?: number;
}
